beoordelingstool_dockwidget.py

```diff
@@ -57,10 +130,722 @@ def add_herstelmaatregelen(self):
+            self.tablewidget_pipes.setItem(0, 40, QTableWidgetItem(f["AXA"]))
+            self.tablewidget_pipes.setItem(0, 41, QTableWidgetItem(f["AXB"]))
+            self.tablewidget_pipes.setItem(0, 42, QTableWidgetItem(f["AXF"]))
+            self.tablewidget_pipes.setItem(0, 43, QTableWidgetItem(f["AXG"]))
+            self.tablewidget_pipes.setItem(0, 44, QTableWidgetItem(f["AXH"]))
+            self.selected_pipe_id = f.id()
+
+    def save_beoordeling_leidingen(self):
+        """Save herstelmaatregel and opmerking in the shapefile."""
+        layer = iface.activeLayer()
+        fid = self.selected_pipe_id
+        herstelmaatregel = str(self.field_combobox_pipes.currentText())
+        opmerking = str(self.value_plaintextedit_pipes.toPlainText())
+        layer.startEditing()
+        layer.changeAttributeValue(fid, 46, herstelmaatregel)  # Herstelmaatregel
+        layer.changeAttributeValue(fid, 47, opmerking)  # Opmerking
+        layer.commitChanges()
+        layer.triggerRepaint()
+
+    def show_measuring_point(self):
+        """Show the measuring station that belongs to a certain pipe."""
+        try:
+            current_measuring_point_pipe_id = self.tablewidget_measuring_points.itemAt(0,0).text()
+        except:  # No measuring point selected
+            current_measuring_point_pipe_id = -1
+        # if current_measuring_point_pipe_id != self.selected_pipe_id:
+        layerList = QgsMapLayerRegistry.instance().mapLayersByName(SHP_NAME_MEASURING_POINTS)
+        if layerList:
+            layer = layerList[0]
+            expr = QgsExpression("\"PIPE_ID\" IS '{}'".format(self.selected_pipe_id))
+            measuring_points = layer.getFeatures(QgsFeatureRequest(expr))  #.next()
+            ids = [measuring_point.id() for measuring_point in measuring_points]  # select only the features for which the expression is true
+            first_id = ids[0]
+            last_id = ids[-1] if ids[-1] else ids[0]
+            # Show selected measuring station if it belongs to the selected pipe
+            if self.selected_measuring_point_id >= first_id and self.selected_measuring_point_id <= last_id:
+                pass
+            # Show first measuring station that belongs to the selected pipe:
+            else:
+                self.selected_measuring_point_id = first_id
+            layer.setSelectedFeatures([int(self.selected_measuring_point_id)])
+            new_feature = layer.selectedFeatures()[0]
+
+            # Set values
+            self.field_combobox_measuring_points.setCurrentIndex(self.field_combobox_measuring_points.findText(str(new_feature["Herstelmaa"]))) if self.field_combobox_measuring_points.findText(str(new_feature["Herstelmaa"])) else self.field_combobox_measuring_points.setCurrentIndex(0)
+            opmerking = new_feature["Opmerking"] if new_feature["Opmerking"] and type(new_feature["Opmerking"]) is not QPyNullVariant else ''
```

Comment:
Ik weet niet of het hier zit, maar: liever niet "NULL" in de widget tonen maar dan een lege string ofzo.

Reply:
Dit lost het probleem op. :)
De lege string van de json wordt vertaald naar een lege string in de shapefile. Als je die in de tabel van de dockwidget will laten zien als een lege string en niet als "NULL", moet je hier kijken of het type geen QPyNullVariant is.